File: advent_of_code/2019/day1/the-tyranny-of-the-rocket-equation.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>

int getFuelMass(int totalMass) {
  return totalMass / 3 - 2;
}

int main() {
  std::ifstream file("day1.txt");

  if (file.is_open()) {
    std::string line;
    int totalFuel = 0;
    int totalAdditionalFuel = 0;

    while(getline(file, line)) {
      int totalMass = std::stoi(line);

      // Part 1
      int fuelMass = getFuelMass(totalMass);
      totalFuel += fuelMass;

      // Part 2
      do {
        totalAdditionalFuel += fuelMass;
      } while ( (fuelMass = getFuelMass(fuelMass)) > 0);
    }

    file.close();

    std::cout << "Part 1: " << totalFuel << std::endl;
    std::cout << "Part 2: " << totalAdditionalFuel << std::endl;
  }
}

File: advent_of_code/2019/day1/the-tyranny-of-the-rocket-equation.py
import os

def get_total_fuel(file: str) -> None:
    with open(file) as f:
        total_fuel = 0
        for line in f:
            total_mass = int(line)
            fuel_mass = total_mass // 3 - 2
            total_fuel += fuel_mass
        return total_fuel

# Part 1
total_fuel = get_total_fuel("day1.txt")
print(total_fuel)
